packages/actions/index.ts

```diff
@@ -1025,26 +1025,43 @@ const settleBilateral = async (user, bilateralSettlements, dispatch, getState) =
   const ethBalance = getState().store.ethBalance
 
   bilateralSettlements.forEach( async (settlement) => {
-    if (settlement.creditorAddress.indexOf(user.address) === -1 || settlement.txHash !== undefined) {
+    if (settlement.creditorAddress.indexOf(user.address) === -1 || !!settlement.txHash) {
       return
     }
 
-    if ( Number(`${settlement.settlementAmount}`) > Number(web3.toWei(ethBalance, 'ether')) ) {
+    let insufficientFunds = false
+    if (settlement.settlementCurrency === 'ETH') {
+      insufficientFunds = Number(`${settlement.settlementAmount}`) > Number(web3.toWei(ethBalance, 'ether'))
+    } else {
+      //assumes ERC20 token
+      const ERC20Token = getERC20_token(settlement.settlementCurrency)
+      const ERC20Balance = await ERC20Token.getBalance(user.address)
+      insufficientFunds = Number(`${settlement.settlementAmount}`) > ( Number(ERC20Balance) * Math.pow(10, ERC20Token.decimals) )
+    }
+
+    if ( insufficientFunds ) {
       const debtorNickname = await getNicknameForAddress(settlement.debtorAddress)
       return dispatch(displayError(settlementManagement.bilateral.error.insufficient(debtorNickname)))
     }
 
-    const ethTransaction = new ERC20_Transaction(settlement.creditorAddress, settlement.debtorAddress, settlement.settlementAmount, gasPrice, GAS_TO_SETTLE_WITH_ETH)
+    const gasNeeded = (settlement.settlementCurrency.toUpperCase() === 'ETH') ? GAS_TO_SETTLE_WITH_ETH : GAS_TO_SEND_ERC20
```

Comment:
We compare `settlement.settlementCurrency` 4 times in this file, yet we only convert it to uppercase once, in this case above. We should declare a local const variable at the top and convert to uppercase, then use that everywhere for consistency.